src/Microsoft.Identity.Client/Cache/Items/MsalAppMetadataCacheItem.cs

```diff
@@ -0,0 +1,136 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Microsoft.Identity.Client.Cache.Keys;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.Cache.Items
+{
+    /// <summary>
+    /// Apps shouldn't rely on its presence, unless the app itself wrote it. It means that SDK should translate absense of app metadata to the default values of its required fields.
+    /// Other apps that don't support app metadata should never remove existing app metadata.
+    /// App metadata is a non-removable entity.It means there's no need for a public API to remove app metadata, and it shouldn't be removed when removeAccount is called.
+    /// App metadata is a non-secret entity. It means that it cannot store any secret information, like tokens, nor PII, like username etc.
+    /// App metadata can be extended by adding additional fields when required.Absense of any non-required field should translate to default values for those field.
+    /// </summary>
+    internal class MsalAppMetadataCacheItem : MsalItemWithAdditionalFields, IEquatable<MsalAppMetadataCacheItem>
+    {
+        public MsalAppMetadataCacheItem(string clientId, string env, string familyId)
+        {
+            this.ClientId = clientId;
+            this.Environment = env;
+            this.FamilyId = familyId;
+        }
+
+        /// <remarks>mandatory</remarks>
+        public string ClientId { get; }
+
+        /// <remarks>mandatory</remarks>
+
+        public string Environment { get; }
+
+        /// <summary>
+        /// The family id of which this application is part of. This is an internal feature and there is currently a single app,
+        /// with id 1. If familyId is empty, it means an app is not part of a family. A missing entry means unkown status.
+        /// </summary>
+        public string FamilyId { get; }
+
+        public MsalAppMetadataCacheKey GetKey()
+        {
+            return new MsalAppMetadataCacheKey(ClientId, Environment);
+        }
+
+        internal static MsalAppMetadataCacheItem FromJsonString(string json)
+        {
+            return FromJObject(JObject.Parse(json));
+        }
+
+        internal static MsalAppMetadataCacheItem FromJObject(JObject j)
+        {
+            string clientId = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.ClientId);
+            string environment = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.Environment);
+            string familyId = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.FamilyId);
+
+            var item = new MsalAppMetadataCacheItem(clientId, environment, familyId);
```

Comment:
why not have a ctror or factory method on the MsalApMetadata... object which takes the JObject?

Reply:
There is one lower down - internal static MsalAppMetadataCacheItem FromJObject(JObject j)